### ISSUE TITLE
AutoLoginGuard does not start token refresh

### ISSUE THREAD

**Issue by thatsmeta** [damienbod/angular-auth-oidc-client#1176]
**Describe the bug**
Hi guys, very happy with the library so far. I am using Auth Code Flow with PKCE and silent refresh via iframe. My goal is to secure the whole Angular App including the start page and I am using the `AutoLoginGuard` to do so (stuck with Angular 11 and Version 11.6.11 of this library).
However, when using only the AutoLoginGuard, it seems that the token refresh procedure is never initialized. No iframe is created. If I put an additional `oidcSecurityService.checkAuth()` in the `AppComponent`, refresh works, but then I get two token requests in my app after the login redirect and one of them fails with a 400 because it tries to use the same auth code as the first one.

Can you help me how to fix this?
My goal is to secure my whole app while supporting deep linking and automatic token refresh.

**To Reproduce**
1. Configure auth with Code Flow, PKCE and silent renew via iframe
2. Use AutoLoginGuard for home route
2. Login
3. No iframe appears, token refresh is not working

**Expected behavior**
AutoLoginGuard should initialize the token refresh procedure

**Desktop (please complete the following information):**
 - OS: Mac OS
 - Browser Firefox, Chrome
 - Version 89, 93

**Comment by bas-be** [comment]
I experienced the same issue and did a short investigation.

Setting the config property _triggerAuthorizationResultEvent: true_ fixes that the token refresh procedure is never initialized.

https://github.com/damienbod/angular-auth-oidc-client/blob/26bd4bb7d0ec04f996c8b88043991da07102e7c7/projects/angular-auth-oidc-client/src/lib/callback/code-flow-callback.service.ts#L26-L31

Unfortunately this instroduces many redirects in my application.
Therefore I did a bit of debugging and the problem seems to be that after the tokens are received and persisted the next for the callback observable is never fired and therefore startCheckSessionAndValidation is never called.

https://github.com/damienbod/angular-auth-oidc-client/blob/26bd4bb7d0ec04f996c8b88043991da07102e7c7/projects/angular-auth-oidc-client/src/lib/check-auth.service.ts#L135-L158
